std/format.d

```diff
@@ -3609,6 +3609,11 @@ if (is(T == class) && !is(T == enum))
     // TODO: Change this once toString() works for shared objects.
     static assert(!is(T == shared), "unable to format shared objects");
 
+    // TODO: remove this check once `@disable override` deprecation cycle is finished
+    static if (__traits(hasMember, T, "toString") && isSomeFunction!(val.toString))
```

Comment:
There's a flaw with this test: if a struct defines `@disable toString` but it isn't a function, `isSomeFunction!(val.toString)` will not compile (this appears to be a bug in `std.traits` or the compiler, will investigate further).  Furthermore, if `toString` is not disabled but also isn't a function, `__traits(isDisabled)` will still work.  So `isSomeFunction` really isn't necessary here.

Reply:
Actually, correction: `__traits(isDisabled)` will *not* return the correct result if `toString` is not a function but annotated with `@disable`.  This is a compiler bug.

Reply:
the trait return false for any declaration that's not a function.

Reply:
There's a chicken-and-egg problem here.  A struct member that's declared `@disable` cannot even be referenced without producing a compile error, even if passed directly to a `is(T.stringof == function)` test. So there's no way to check whether the symbol is even a function to begin with, and yet `__traits(isDisabled)` returns false for it.  So basically there's no way to avoid a compile error.

Reply:
On second thoughts, the proposed change isn't making things any worse than they are now (compile error with `@disabled`). We can handle the remaining case of declaring `@disable toString` as a non-function later, since it is so rare that it's probably not worth dealing with. The proposed change is bringing *some* improvement, so let's not let the perfect be the enemy of the good.